packages/run-protocol/src/runStake/runStakeKit.js

```diff
@@ -0,0 +1,351 @@
+// @ts-check
+import { Far } from '@endo/far';
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+import { ceilMultiplyBy } from '@agoric/zoe/src/contractSupport/ratio.js';
+import { makeNotifierKit } from '@agoric/notifier';
+import { M, matches } from '@agoric/store';
+import { makeTracer } from '../makeTracer.js';
+import { applyDelta, assertOnlyKeys, transfer } from '../contractSupport.js';
+import { calculateCurrentDebt, reverseInterest } from '../interest-math.js';
+
+const { details: X, quote: q } = assert;
+
+const trace = makeTracer('R1');
+
+const VaultPhase = {
+  ACTIVE: /** @type {'Active'} */ ('Active'),
+  CLOSED: /** @type {'Closed'} */ ('Closed'),
+};
+
+/**
+ * @typedef {VaultPhase[keyof VaultPhase]} VaultPhaseValue
+ * @type {{[K in VaultPhaseValue]: Array<VaultPhaseValue>}}
+ */
+const validTransitions = {
+  [VaultPhase.ACTIVE]: [VaultPhase.CLOSED],
+  [VaultPhase.CLOSED]: [],
+};
+
+/**
+ * Make RUNstake kit, subject to policy.
+ *
+ * @param {ZCF} zcf
+ * @param {ZCFSeat} startSeat
+ * @param {ReturnType<import('./runStakeManager.js').makeRunStakeManager>} manager
```

Comment:
consider defining `RunStakeManager` in its file and importing that.

[this may become standard practice](https://docs.google.com/document/d/1bRIUeOokKcbIHZHt68d9kyiJOB5YEUQVpLpFAmDLkhk/edit#heading=h.6lmefd8kl5rm) for `makeFoo` functions.